src/main/java/emu/grasscutter/command/commands/AnnounceCommand.java

```diff
@@ -0,0 +1,60 @@
+package emu.grasscutter.command.commands;
+
+import emu.grasscutter.Grasscutter;
+import emu.grasscutter.command.Command;
+import emu.grasscutter.command.CommandHandler;
+import emu.grasscutter.game.player.Player;
+import emu.grasscutter.server.packet.send.PacketServerAnnounceNotify;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+import static emu.grasscutter.utils.Language.translate;
+
+@Command(label = "announce",
+    usage = "a <tpl templateId|refresh|revoke templateId|content>",
+    permission = "server.announce",
+    aliases = {"a"},
+    description = "commands.announce.description",
+    targetRequirement = Command.TargetRequirement.NONE)
+public final class AnnounceCommand implements CommandHandler {
+
+    @Override
+    public void execute(Player sender, Player targetPlayer, List<String> args) {
+        var manager = Grasscutter.getGameServer().getAnnouncementManager();
+
```

Comment:
```suggestion
        if (args.size() < 1) {
            CommandHandler.sendTranslatedMessage(sender, "commands.announce.command_usage"));
            return;
        }
```

`"tpl"` and `"revoke"` will need argument error handling too

Reply:
i see